src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -177,14 +177,13 @@ public final void close() throws SQLException
    {
       if (delegate != ClosedConnection.CLOSED_CONNECTION) {
          leakTask.cancel();
+         closeStatements();
```

Comment:
Why this move?  If closeStatements() throws a RuntimeException, or some subclass of it, the finally will not be executed, resulting in the connection not being released back to the pool.